pkg/services/navtree/navtreeimpl/navtree.go

```diff
@@ -362,7 +362,7 @@ func (s *ServiceImpl) buildDashboardNavLinks(c *contextmodel.ReqContext) []*navt
 			})
 		}
 
-		if s.cfg.SnapshotEnabled {
+		if s.cfg.SnapshotEnabled && hasAccess(ac.EvalPermission(dashboards.ActionSnapshotsRead)) {
```

Comment:
What if public snapshots are enabled? Do we need to check the permission in that case?

Reply:
yes, the idea here is, when Snapshots are enabled on the instance (at instance config level) we only show the item menu if the user has Snapshots read access to list the Snapshots.